guest/api/clusterevent.go

```diff
@@ -0,0 +1,77 @@
+package api
+
+import "strconv"
+
+// ActionType is an integer to represent one type of resource change.
+// Different ActionTypes can be bit-wised to compose new semantics.
+type ActionType uint32
+
+// Constants for ActionTypes.
+const (
+	Add    ActionType = 1 << iota // 1
+	Delete                        // 10
+	// UpdateNodeXYZ is only applicable for Node events.
+	UpdateNodeAllocatable // 100
+	UpdateNodeLabel       // 1000
+	UpdateNodeTaint       // 10000
+	UpdateNodeCondition   // 100000
+
+	All ActionType = 1<<iota - 1 // 111111
+
+	// Use the general Update type if you don't either know or care the specific sub-Update type to use.
+	Update = UpdateNodeAllocatable | UpdateNodeLabel | UpdateNodeTaint | UpdateNodeCondition
+)
+
+// GVK is short for group/version/kind, which can uniquely represent a particular API resource.
+type GVK uint32
+
+// Constants for GVKs.
+const (
+	Pod GVK = iota
```

Comment:
after I moved the other types to another package, I was able to mostly copy/paste this. I didn't make a RATIONALE entry about coercion from string to uint32, but on approval I will. The main thing is that this makes encoding far simpler. We should add documentation for `framework.GVK` that custom values are not supported: it is only controlled by kubernetes scheduler framework.